web/admin/src/App.js

```diff
@@ -12,154 +10,530 @@ fbc.initializeAppWithSimpleBackend()
+            question.key === data.key ?
+            { ...question, myVote: userVote, score: question.score - 1}
+            : 
+            question
+          )
+          this.setState({questions})
+        })
       })
-      sharedRef.on('child_removed', data => {
-        this.setState({ sharedVotes: this.state.sharedVotes.filter(x => x.key !== data.key) })
+      questionsRef.on('child_removed', data => {
+        this.setState({ questions: this.state.questions.filter(x => x.key !== data.key) })
       })
-      sharedRefVotes.on('child_changed', data => {
-        this.setState({ sharedVotes: [...this.state.sharedVotes, {...data.val(), key: data.key }] })
+      questionsRef.on('child_changed', data => {
+        var questions = this.state.questions
+        for (var i in questions){
+          if (questions[i].key === data.key) {
+            questions[i] = data.val()
+            questions[i].key = data.key
+            this.setState({questions})
+          }
+        }
       })
     })
   }
 
-  openModal() {
-    this.setState({openVar: true});
-  }
-
-  afterOpenModal() {
-    // references are now sync'd and can be accessed.
-    // this.subtitle.style.color = '#f00';
-    // ()=>this.deleteAll(questions)
-  }
+  render() {
+    const { questions, sharedVotes, showRecent, newUpdate, dropDown, newValue, height, marginTop, moderator} = this.state 
+    const time = new Date().getTime()
+    // var totalQuestions = questions.filter(item => item.approve === false && item.new === true)
+    // if (totalQuestions === undefined){
+    //   totalQuestions = ['']
+    // }
+    questions.sort(function (a,b){
+      return b.dateCreate - a.dateCreate
+    })
+    return (
+    <div className="App">
+      {this.renderModal(questions)}
+      <div className="topBox">
+        <text className='bigBoxTitle'>{'Q & A'}</text>
+        <text className='boxTitle'>Moderation:  </text>
+        {this.renderModIcons()}
+      </div>
+      <div className="container">
+        {/* <div className="questionContainer">
+          <text className='boxTitle'>New Questions: {totalQuestions.length}</text>
+          <span className="questionBox"> */}
+            {this.renderHeader(questions, time)}
+            
+          {/* </span>
+        </div> */}
+        {this.renderRight(questions, time)}
+      </div>
+    
+      
 
-  closeModal() {
-    this.setState({openVar: false});
-    // ()=>this.deleteAll(questions)
+    </div>
 
+    )
   }
 
-  render() {
 
-    const { sharedTasks, sharedVotes } = this.state
-    
-       var questions = sharedTasks.map(t => ({...t, type:'shared'}))
-      //  var questions2 = sharedTasks.map(t => ({...t, type:'shared'}))
-       var votes = sharedVotes.map(t => ({...t, type:'shared'}))
-       var questionScore = questions.map(function(x) { return Object.assign({id: x.key, score: 0}) })
-       
-   
-       this.findScore(questionScore, votes, questions)
+  doDateMath = (date, time) => {
```

Comment:
I think this human-friendly "... ago" text can be generated easily with the `moment` module
http://momentjs.com/docs/#/displaying/fromnow/
https://www.npmjs.com/package/moment